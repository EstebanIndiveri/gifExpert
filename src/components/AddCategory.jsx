import React, { Fragment,useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {

    const[inputValue,SetInputValue]=useState('');

    const inputChange=(e)=>{
        // console.log(e.target.value);
        SetInputValue(e.target.value);
    }

    const handelSubmit=(e)=>{
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategories(cats=>[inputValue,...cats])
            SetInputValue('');
        }
    }

    return ( 
        <Fragment>
        {/* <h1>{inputValue}</h1> */}
        <form onSubmit={handelSubmit}>
            <p>{inputValue}</p>
            <input
            type="text"
            value={inputValue}
            onChange={inputChange}
            />
            </form>
        </Fragment>
     );
}

AddCategory.propTypes={
    setCategories:PropTypes.func.isRequired
}
 
export default AddCategory;