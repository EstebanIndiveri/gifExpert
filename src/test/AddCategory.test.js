import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../components/AddCategory';

describe('pruba en <AddCategody/>', () => {
    const setCategories=jest.fn();
    let wrapper=shallow(<AddCategory setCategories={setCategories}/>);
    beforeEach(()=>{
        jest.clearAllMocks();
    wrapper=shallow(<AddCategory setCategories={setCategories}/>);

    });

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    });
    test('cambiar caja de texto', () => {
        const input=wrapper.find('input');
        const value='hola mundo';
        input.simulate('change',{target:{value:value}});
        expect(wrapper.find('p').text().trim()).toBe(value);

    });

    test('no debe de postear info en el submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
    test('debe de llamar el setCategories y limÃ­ar caja', () => {
   
    //     //1simular inputchange
    //     //2.simular el sumbit
    //     //3.setcategories llamado
    //     //4.input =''
        const value='hola mundo';
        wrapper.find('input').simulate('change',{target:{value}});

        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');
    });
    
})
