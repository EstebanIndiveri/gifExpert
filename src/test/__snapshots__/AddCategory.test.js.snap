// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruba en <AddCategody/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
